catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/transformer/transformers/sierra/SierraDescription.scala

```diff
@@ -0,0 +1,36 @@
+package uk.ac.wellcome.transformer.transformers.sierra
+
+import uk.ac.wellcome.transformer.source.{MarcSubfield, SierraBibData}
+
+trait SierraDescription {
+
+  def getSubfield(
+    bibData: SierraBibData,
+    marcTag: String,
+    marcSubfieldTag: String
+  ): Option[MarcSubfield] = {
+    val matchingFields = bibData.varFields
+      .filter {
+        _.marcTag.contains(marcTag)
+      }
+
+    val matchingSubfields =
+      matchingFields.flatMap {
+        _.subfields
+      }.flatten
+
+    matchingSubfields.find(_.tag == marcSubfieldTag)
+  }
+
+  def getDescription(bibData: SierraBibData): Option[String] = {
+    val descriptionField = getSubfield(bibData, "520", "a")
+    val summaryDescriptionField = getSubfield(bibData, "520", "b")
```

Comment:
Have you checked that we would only ever have a single instance of these subfields on the entire MARC record?

Reply:
Yes, apparently there can be!

Reply:
err…?

Can be multiple, or can only be one? 😄